jupyterhub/spawner.py

```diff
@@ -703,10 +703,11 @@ def run_pre_spawn_hook(self):
 
     def run_post_stop_hook(self):
         """Run the post_stop_hook if defined"""
-        try:
-            return self.post_stop_hook(self)
-        except Exception:
-            self.log.exception("post_stop_hook failed with exception: %s", self)
+        if self.post_stop_hook:
```

Comment:
nitpick: can we explicitly check for `None` with `if self.post_stop_hook is not None:`?